### PR TITLE
Add support for creating directories that don't exist on renderFiles

### LINKED ISSUE #247: writeRenderedFile() should create the destination directories
**Enhancement** 

Both `@dojo/cli-create-app` and `@dojo/cli-create-widget` have to manually create the directories they are rendering into before calling `helper.command.renderFiles()`. This should be done automatically by `writeRenderedFiles()`.

Package Version: 3.0.0-alpha.1

### PR DESCRIPTION
**Pull request by JamesMilnerUK**
**Type:** Feature

The following has been addressed in the PR:

* [x] There is a related issue
* [x] All code has been formatted with [`prettier`](https://prettier.io/) as per the [readme code style guidelines](./../#code-style)
* [x] Unit or Functional tests are included in the PR

**Description:**

Resolves #247

### CODE REVIEW

File: src/template.ts [hunk 1]
```diff
@@ -15,6 +16,8 @@ export function ejsRender(source: string, replacements: Object): Promise<string>
 
 export function writeRenderedFile(str: string, destination: string): Promise<void> {
 	return new Promise<void>((resolve, reject) => {
+		console.log('?????? \n');
```
Comment (bryanforbes) [inline src/template.ts]: This should be removed

File: src/template.ts [hunk 2]
```diff
@@ -15,6 +16,8 @@ export function ejsRender(source: string, replacements: Object): Promise<string>
...
+		ensureDirectoryExistence(destination);
```
Comment (bryanforbes) [inline src/template.ts]: Why not use [`fs-extra.ensureDir()`](https://github.com/jprichardson/node-fs-extra/blob/master/docs/ensureDir.md)?

File: src/template.ts [hunk 3]
```diff
@@ -15,6 +15,7 @@ export function ejsRender(source: string, replacements: Object): Promise<string>
 
 export function writeRenderedFile(str: string, destination: string): Promise<void> {
 	return new Promise<void>((resolve, reject) => {
+		ensureDir(destination);
```
Comment (bryanforbes) [inline src/template.ts]: `ensureDir()` is an asynchronous function and can either take a callback (like `writeFile()`) or returns a promise.

Comment (JamesMilnerUK) [reply src/template.ts]: Ahh yes.

File: src/template.ts [hunk 4]
```diff
@@ -1,26 +1,22 @@
...
 
 export function writeRenderedFile(str: string, destination: string): Promise<void> {
-	return new Promise<void>((resolve, reject) => {
-		writeFile(destination, str, (err?: Error) => {
-			if (err) {
-				reject(err);
-			}
-			resolve();
-		});
+	return ensureDir(destination).then(() => {
```
Comment (bryanforbes) [inline src/template.ts]: Just for sanity's sake, can you make sure that both of these functions return a promise in the version of `fs-extra` we have pinned?

Comment (JamesMilnerUK) [reply src/template.ts]: For sure!

Comment (JamesMilnerUK) [reply src/template.ts]: @bryanforbes looks like they do:

screenshot_20180904_171506

Comment (JamesMilnerUK) [reply src/template.ts]: and writeFile does also:
screenshot_20180904_180337

**Pull request merged**